File: demo/src/main/java/com/example/demo/dto/UserSearchDto.java
package com.example.demo.dto;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UserSearchDto {
    private String query;
    private String city;
    private String profession;
    private String status;

    private Integer ageFrom;
    private Integer ageTo;

    private Integer experienceFrom;
    private Integer experienceTo;
}

File: demo/frontend/user-management/src/app/Users/interfaces/dto/user-search.dto.ts
export interface UserSearchDto {
  query?: string;
  city?: string;
  profession?: string;
  status?: 'ACTIVE' | 'INACTIVE' | 'SUSPENDED';

  ageFrom?: number;
  ageTo?: number;

  experienceFrom?: number;
  experienceTo?: number;
}
